### ISSUE TITLE
[Feature Request] Custom field groups

### ISSUE THREAD

**Issue by joshleblanc** [deepnight/ldtk#1101]
We can define entities in LDTK right now, and we can define custom fields on entities.

If we could create groups of custom fields called "components", we could implement an ECS system in the codebase simply by implementing systems and allowing LDTK to define the components and entities.

The behavior of a component would be similar to the behavior of a single custom field, given a new tab at the top to define them. For example, you define a component "bumpable" and it has the custom fields "power" and "loot".  Now you can add that component to any entity, and it would contain power and loot as custom fields under "bumpable". The values default to the value from the component, but can be overridden.

Basically just a way to group re-usable custom fields to place on entities.
